src/swap/components/SwapProvider.tsx

```diff
@@ -163,7 +166,8 @@ export function SwapProvider({
 
   const handleSubmit = useCallback(
     async function handleSubmit(
-      onSubmit?: (swapTransaction: BuildSwapTransaction) => void,
+      onError?: (error: SwapError) => void,
+      onSuccess?: (txReceipt: TransactionReceipt) => void | Promise<void>,
```

Comment:
asking out loud, should `txReceipt` just be `transactionReceipt`? just for better clarity.